fix!: prevent creation of chips on keyboard events with modifier

beawar opened the pull request:
Increase the control over keyboard separators of the ChipInput by defining the separators as keyboard-event-like objects. Using a string and checking the match with the key or the code of the keyboard event, was causing the creation of the chips even when the match was with the code, but there was an active modifier. The problem was emphasized on keyboard layouts differents from the US one.

BREAKING CHANGE: update the keyboard utils to accept only event-like keys In order to migrate from the old string keys to the new event-like keys, transform each string in an object with the previous string set as key or code (according to its value), and the field ctrlKey set to the value of the old "modifier" field/argument (its default was false).
BREAKING CHANGE: update the ChipInput separators prop to receive only event-like keys

refs: CDS-98

beawar committed:
fix!: prevent creation of chips on keyboard events with modifier

Increase the control over keyboard separators of the ChipInput
by defining the separators as keyboard-event-like objects.
Using a string and checking the match with the key or the code
of the keyboard event, was causing the creation of the chips even when
the match was with the code, but there was an active modifier.
The problem was emphasized on keyboard layouts differents from the US one.

BREAKING CHANGE: update the keyboard utils to accept only event-like keys
In order to migrate from the old string keys to the new event-like keys,
transform each string in an object with the previous string set as key or code
(according to its value), and the field ctrlKey set to the value of the old "modifier"
field/argument (its default was false).
BREAKING CHANGE: update the ChipInput separators prop to receive only event-like keys

refs: CDS-98

CataldoMazzilli commented on src/hooks/useKeyboard.tsx:
```diff
@@ -212,13 +207,9 @@ function useKeyboard(ref: React.RefObject<HTMLElement>, events: KeyboardPreset):
 		() =>
```
update function comment

beawar committed:
fix(types)!: remove KeyboardPreset type

KeyboardPreset can be replaced with KeyboardPresetObj[].
It was redundant and making some typing more confusing.

refactor: rename useKeyboard args to make the code more readable

Update the description of the hook to remove the empty string escamotage.

BREAKING CHANGE: remove KeyboardPreset type
refs: CDS-98

beawar committed:
Merge remote-tracking branch 'origin/devel' into CDS-98-fix-chipinput-different-keyboard

beawar committed:
Merge remote-tracking branch 'origin/devel' into CDS-98-fix-chipinput-different-keyboard
